### PR TITLE
provider/openstack: update goose

### PR DESCRIPTION
**Pull request by axw**
Use new version of goose, to omit the device
field on volume attachment requests when no
device is specified.

This branch also uses the new OpenStack API
version discovery code in goose.

Fixes https://bugs.launchpad.net/juju/+bug/1636648

### CODE REVIEW

**Comment by axw** [comment]
**QA**

1. bootstrap canonistack/lcy01
2. add-machine --disks cinder,1G

File: provider/openstack/cinder.go [hunk 1]
```diff
@@ -381,11 +381,14 @@ func (s *cinderVolumeSource) attachVolume(arg storage.VolumeAttachmentParams) (*
 			return nil, err
 		}
 	}
+	if novaAttachment.Device == nil {
+		return nil, errors.Errorf("device not assigned to volume attachment")
```
Comment (wallyworld) [inline provider/openstack/cinder.go]: test for this?

Comment (axw) [inline provider/openstack/cinder.go]: Done.

**Comment by axw** [comment]
$$merge$$

**Comment by axw** [comment]
$$merge$$

**Pull request merged**